File: kotlin-j2ee/src/main/java/edu/learn/web/filter/Interceptor.java
package edu.learn.web.filter;

import javax.servlet.*;
import javax.servlet.annotation.WebFilter;
import java.io.IOException;

@WebFilter("/*")
public class Interceptor implements Filter {
    @Override
    public void init(FilterConfig filterConfig) throws ServletException {
        System.out.println("Init");
    }

    @Override
    public void doFilter(ServletRequest servletRequest, ServletResponse servletResponse, FilterChain filterChain) throws IOException, ServletException {

        filterChain.doFilter(servletRequest, servletResponse);
    }

    @Override
    public void destroy() {
        System.out.println("Destroy");
    }
}

File: kotlin-j2ee/src/main/kotlin/edu/learn/kotlin/web/filter/Interceptor.kt
package edu.learn.kotlin.web.filter

import javax.servlet.*
import javax.servlet.annotation.WebFilter

//@WebFilter(filterName = "Interceptor", urlPatterns = arrayOf("/*"))
@WebFilter("/*")
class Interceptor : Filter {
    override fun destroy() {
        println("Interceptor has been destroy ...")
    }

    override fun doFilter(request: ServletRequest?, response: ServletResponse?, chain: FilterChain?) {
        println("Hey ...")
        chain?.doFilter(request, response)
    }

    override fun init(p0: FilterConfig?) {
        println("Interceptor has been initialized ...")
    }
}